src/js/component.js

```diff
@@ -1524,6 +1498,53 @@ class Component {
     return id;
   }
 
+  /**
+   * Request an animation frame, but only one named animation
+   * frame will be queued. Another will never be added until
+   * the previous one finishes.
+   *
+   * @param {string} name
+   *        The name to give this requestAnimationFrame
+   *
+   * @param  {Component~GenericCallback} fn
+   *         A function that will be bound to this component and executed just
+   *         before the browser's next repaint.
+   */
+  requestNamedAnimationFrame(name, fn) {
+    if (this.namedRafs_.has(name)) {
+      return;
+    }
+    this.clearTimersOnDispose_();
+
+    fn = Fn.bind(this, fn);
+
+    const id = this.requestAnimationFrame(() => {
+      fn();
+      if (this.namedRafs_.has(name)) {
+        this.namedRafs_.delete(name);
+      }
```

Comment:
I noticed that rAF has a similar implementation on lines 1491-1494 but it first deletes the id from the rafIds before calling the method. Not sure if the ordering matters much. Calling the function first sounds ok for me.

Reply:
I think we delete it afterwards because we want to make sure that the entire update runs before we try to queue another one. If an update takes awhile we could delete the animation frame and then queue another one before that update finishes.

Reply:
I added assertions to the unit test for this function so that we test for this.

Reply:
ok, makes sense.